Mathlib/GroupTheory/OrderOfElement.lean

```diff
@@ -711,6 +718,29 @@ theorem IsOfFinOrder.mul (hx : IsOfFinOrder x) (hy : IsOfFinOrder y) : IsOfFinOr
 
 end CommMonoid
 
+section CommGroup
+variable [CommGroup G]
+
+@[to_additive]
+lemma isMulTorsionFree_iff_not_isOfFinOrder :
```

Comment:
Doesn't `[injective_pow_iff_not_isOfFinOrder](https://leanprover-community.github.io/mathlib4_docs/Mathlib/GroupTheory/OrderOfElement.html#injective_pow_iff_not_isOfFinOrder)` help you prove this?

Reply:
Nope. I originally thought the same, but `injective_pow_iff_not_isOfFinOrder` talks about `(x ^ ·)` while `IsMulTorsionFree` talks about `(· ^ n)`. It's quite funny to me that the injectivity of both functions are related